### DISCUSSION TITLE
Establishing a verfiable track record

### DISCUSSION THREAD

**Discussion by SunnySideUpwards** [community/community#21456] [Other Feature Feedback, Questions, & Ideas]
<p>Hello everyone</p>
<p>My apololgies for re-posting this (as I am completely new to GitHub and have only just found this). I am completely new to GitHub and I hope I have come to the right place, as this site was recommended to me. I am not a programmer (but know someone who is). My problem is more of a coneceptual one and would greatly appreciate help or ideas to what I regard as a “Catch 22” situation.</p>
<p>I have developed a hedge fund model that generates recommendations on long and short positions on a daily basis. I now need to prove that it works by submitting a recommendation to some address/ platform, i.e. establish a track record.</p>
<p>The catch is that in order to make this verifiable, I should not be allowed to make changes to the recommendations less than 40 minutes before the market in Wall Street opens (09:30 EST) on that given day. As I would presumably be the administrator of such a construction, how would I verifiably keep myself from doing so or in other words credibly, in the eyes of prospect investors, keep myself from “cheating”?</p>
<p>I imagine there are many great investment ideas out there that cannot access funding (from investors) for the same reason!</p>
<p>Any ideas of this – even a general sketch in the way of a set-up - would be greatly appreciated!</p>
<p>Kind regards,</p>
<p>SunnySideUpwards</p>

**Comment by mpboom** [comment]
<p>Store your predictions on some public blockchain beforehand. If you pick a network large enough, the people you’re trying to convince can be very sure you didn’t play dirty.</p>

**Comment by SunnySideUpwards** [comment]
<p>Hi mpboom, Many thanks for your idea. This would be completely new to me. Could you please elaborate on where I could go to and what the basic steps would look like? I would alos need to calculate the perfromance based on these recommendations. Would that also be possible there? Greetings, SunnySideUpwards</p>

**Comment by SunnySideUpwards** [comment]
<p>Hi mpboom, One question here: Would this then be “public” information, as you mention a network that is “large enough”? I would need these recomemdations to remain confidential or restrict the acces in some way. Any ideas on this?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>In general, what you need a trusted timestamp. Basically a combination of your document with time information, signed by someone else who the people who might want to verify your record trust (the equivalent of a notary). Putting the document (or a hash of it, to avoid publishing the whole thing) on a blockchain relies on it being extremely hard to manipulate once a block is generally accepted.</p>
<p>The Wikipedia article on the matter describes the concept and mentions a bunch of implementations:</p>
<aside class="onebox wikipedia">
  <header class="source">
      <a href="https://en.wikipedia.org/wiki/Trusted_timestamping" target="_blank" rel="noopener">en.wikipedia.org</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:330/201;"><img src="https://user-images.githubusercontent.com/5814919/181064633-3ac72bcf-ecc0-4970-a0ed-9b4aad03c3ec.png" class="thumbnail" width="330" height="201"></div>

<h3><a href="https://en.wikipedia.org/wiki/Trusted_timestamping" target="_blank" rel="noopener">Trusted timestamping</a></h3>

<p>Trusted timestamping is the process of securely keeping track of the creation and modification time of a document. Security here means that no one—not even the owner of the document—should be able to change it once it has been recorded provided that the timestamper's integrity is never compromised.
 The administrative aspect involves setting up a publicly available, trusted timestamp management infrastructure to collect, process and renew timestamps.
 The idea of timestamping information is centu...</p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by SunnySideUpwards** [comment]
<p>Hi airtower-luna, Yes I think you got the idea and I will look into this, thank you. One crucial thing is, that I need to keep this info confidential i.e. limit a users ability to view this only if I want them to. Is this possible (if not, I’ll have to stop right here)? Greetings, SunySideUpwards</p>

**Comment by airtower-luna** [comment]
<p>That’s why I mentioned hashing the document. In practice signature algorithms involve hashing and signing the hash anyway.</p>

**Comment by SunnySideUpwards** [comment]
<p>Hi airtower-luna,<br>
Justr wanted to thank you for putting me on the right track! It turns out that there are trusted sites that you can connect with via a standard Adobe pdf reader document (via the saftey settings) and have it certified with a trusted time stamp. Very cool, thank you.<br>
My next step will now be to automate this procedure through some kind of macro. Since this is a new issue I will open a new thread (just so you know and the topic sounds familiar :smile:)</p>
